src/Conversions/include/BipedalLocomotion/Conversions/ManifConversions.h

```diff
@@ -0,0 +1,113 @@
+/**
+ * @file Conversions.h
+ * @authors Prashanth Ramadoss
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_MANIF_CONVERSIONS_H
+#define BIPEDAL_LOCOMOTION_MANIF_CONVERSIONS_H
+
+#include <iDynTree/Core/Transform.h>
+#include <iDynTree/Core/EigenHelpers.h>
+#include <manif/manif.h>
+
+
+namespace BipedalLocomotion
+{
+namespace Conversions
+{
+    /**
+     * @brief Convert rotation matrix and translation vector to manif SE3 object
+     *
+     * @param rotation reference to 3x3 Eigen matrix
+     * @param translation reference of 3x1 Eigen matrix
+     * @param pose reference to manif SE3 object
+     */
+    template <class Scalar>
+    void eigenRotTrans2ManifPose(const Eigen::Matrix<Scalar, 3, 3>& rotation,
+                                 const Eigen::Matrix<Scalar, 3, 1>& translation,
+                                 manif::SE3<Scalar>& pose)
+    {
+        Eigen::Quaternion<Scalar> quat = Eigen::Quaternion<Scalar>(rotation);
+        quat.normalize(); // SO3 constructor expects normalized quaternion
+        pose = manif::SE3<Scalar>(translation, quat);
+    }
+
+    /**
+     * @brief Convert rotation matrix and translation vector to manif SE3 object
+     *
+     * @param rotation reference to 3x3 Eigen matrix
+     * @param translation reference of 3x1 Eigen matrix
+     * @return pose as manif SE3 object
+     */
+    template <class Scalar>
+    manif::SE3<Scalar> eigenRotTrans2ManifPose(const Eigen::Matrix<Scalar, 3, 3>& rotation,
+                                               const Eigen::Matrix<Scalar, 3, 1>& translation)
+    {
+        manif::SE3<Scalar> pose;
+        eigenRotTrans2ManifPose(rotation, translation, pose);
+        return pose;
+    }
+
+    /**
+     * @brief Convert rotation matrix and translation vector to manif SE3d object
+     *
+     * @param rotation Eigen ref of 3x3 rotation matrix
+     * @param translation Eigen ref of 3x1 translation vector
+     * @param pose reference to manif SE3d object
+     */
+    inline void eigenRotTrans2ManifPose(Eigen::Ref<const Eigen::Matrix3d> rotation,
+                                        Eigen::Ref<const Eigen::Vector3d> translation,
+                                        manif::SE3d& pose)
+    {
+        Eigen::Quaterniond quat = Eigen::Quaterniond(rotation);
+        quat.normalize(); // SO3 constructor expects normalized quaternion
+        pose = manif::SE3d(translation, quat);
+    }
+
```

Comment:
Is this version really needed? It seems a duplicate of the one below

Reply:
I was reasoning the user instead of creating a new object through return by value can maybe pass to overwrite a really old existing object?

Reply:
Since the dimension is fixed, it is not a big deal. In addition, if manif implements the move assignment, also the one returning directly the value should be efficient. In addition, the other version is preferable if you don't have a manif object

Reply:
Ok, I shall remove these methods then. With `manif` the storage of pose is 3+7, so it's not too heavy.


But should we also apply this change also to the CommonConversions method dealing with Matrix4d?

Reply:
Yes, that would be great

Reply:
Addressed in https://github.com/dic-iit/bipedal-locomotion-framework/commit/b6eadd6b205264568dfb418808f37da03bdbde8e